bftengine/src/bftengine/ReplicaForStateTransfer.cpp

```diff
@@ -94,12 +123,11 @@ void ReplicaForStateTransfer::start() {
                 succ = false;
                 break;
               }  // else if (!isGettingBlocks)
+              // 2. Now we can safely halt cre. We know for sure that there are no update in the state transferred
+              // blocks that haven't been handled yet
+              cre_->halt();
```

Comment:
I think you need to move this outside of the while loop. 
Here you simply halt the cre on the very first loop.